internal/gatewayapi/testdata/httproute-with-redirect-filter-invalid-status.out.yaml

```diff
@@ -0,0 +1,99 @@
+gateways:
+  - apiVersion: gateway.networking.k8s.io/v1beta1
+    kind: Gateway
+    metadata:
+      namespace: envoy-gateway
+      name: gateway-1
+    spec:
+      gatewayClassName: envoy-gateway-class
+      listeners:
+        - name: http
+          protocol: HTTP
+          port: 80
+          hostname: "*.envoyproxy.io"
+          allowedRoutes:
+            namespaces:
+              from: All
+    status:
+      listeners:
+        - name: http
+          supportedKinds:
+            - group: gateway.networking.k8s.io
+              kind: HTTPRoute
+          attachedRoutes: 1
+          conditions:
+            - type: Ready
+              status: "True"
+              reason: Ready
+              message: Listener is ready
+httpRoutes:
+  - apiVersion: gateway.networking.k8s.io/v1beta1
+    kind: HTTPRoute
+    metadata:
+      namespace: default
+      name: httproute-1
+    spec:
+      hostnames:
+      - gateway.envoyproxy.io
+      parentRefs:
+        - namespace: envoy-gateway
+          name: gateway-1
+          sectionName: http
+      rules:
+        - matches:
+            - path:
+                value: "/"
+          backendRefs:
+            - name: service-1
+              port: 8080
+          filters:
+          - type: RequestRedirect
+            requestRedirect:
+              scheme: https
+              statusCode: 666
+    status:
+      parents:
+        - parentRef:
+            namespace: envoy-gateway
+            name: gateway-1
+            sectionName: http
+          # controllerName: envoyproxy.io/gateway-controller
+          conditions:
+            - type: Accepted
+              status: "True"
+              reason: Accepted
+              message: Route is accepted
+            - type: ResolvedRefs
+              status: "False"
+              reason: UnsupportedValue
+              message: "Status code 666 is invalid, only 302 and 301 are supported"
+xdsIR:
+  http:
+    - name: envoy-gateway-gateway-1-http
+      address: 0.0.0.0
+      port: 80
+      hostnames:
+      - "*.envoyproxy.io"
+      routes:
+        - name: default-httproute-1-rule-0-match-0-gateway.envoyproxy.io
+          pathMatch:
+            prefix: "/"
+          headerMatches:
+            - name: ":authority"
+              exact: gateway.envoyproxy.io
+          # I believe the correct way to handle an invalid filter should be to allow the HTTPRoute to function
+          # normally but leave out the filter config and set the status, but this behaviour can be changed.
```

Comment:
Yeah, I think the expected behavior here is unclear in the Gateway API spec -- we can drive some clarity upstream, but doesn't need to block this PR now.